File: src/atcoder/abc308/abc308_a.scala
import java.util.Scanner
import scala.collection.immutable.Vector

object Main extends App {
  val sc = new Scanner(System.in)
  val s = Vector.fill(8) { sc.nextInt }

  def solve: Boolean = {
    s.forall(v => 100 <= v && v <= 675 && v % 25 == 0) &&
    (1 to 7).forall(i => s(i - 1) <= s(i))
  }

  println(if (solve) "Yes" else "No") 
}

File: src/atcoder/abc308/kotlin/abc308_a.kt
fun main() {
  val a = readLine()!!.split(" ").map { it.toInt() }
  println(solve(a))
 }

fun solve(a: List<Int>): String {
  val n = a.size
  // 数列 a は広義単調増加である
  for (i in 0..(n-2)) {
    if (i != n - 1 && a[i] > a[i+1]) {
      return "No"
    }

    if (a[i] < 100 || 675 < a[i]) {
      return "No"
    }

    if (a[i] % 25 != 0) {
      return "No"
    }
  }
  return "Yes"
}